lib/Models/Catalog/Ows/WebMapServiceCatalogItem.ts

```diff
@@ -665,12 +767,26 @@ class WebMapServiceCatalogItem
           newStyle: string | undefined
         ) => {
           if (!newStyle) return;
+
           runInAction(() => {
             const styles = this.styleSelectableDimensions.map(
               (style) => style.selectedId || ""
             );
-            styles[layerIndex] = newStyle;
-            this.setTrait(stratumId, "styles", styles.join(","));
+            styles[layerIndex] = newStyle ? newStyle : "";
+
+            if (this.isNcWMS && this.palette) {
+              const currentStyle = newStyle.split("/")[0];
+              const currentPalette = this.palette;
+              const newStyles = currentStyle + "/" + currentPalette;
+              this.setTrait(stratumId, "styles", newStyles);
+              if (this.noPaletteStyles.includes(newStyle)) {
+                this.setTrait(stratumId, "showPalettes", false);
+              } else {
+                this.setTrait(stratumId, "showPalettes", true);
+              }
+            } else {
+              this.setTrait(stratumId, "styles", styles.join(","));
+            }
```

Comment:
Same as above, I think you should keep `styles` and `palette` separate. After creating the `NcWMSLoadableStratum`, you can change `showPalette` there without explicitly calling `setTrait`.